File: backend/src/main/scala/microservices/content/objects/DeleteReviewPlannerRequest.scala
package com.typesafe.travel.content.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class DeleteReviewPlannerRequest(
    userId: String,
    reviewId: String
)
object DeleteReviewPlannerRequest:
  given sourceEncoder: Encoder[DeleteReviewPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[DeleteReviewPlannerRequest] = deriveDecoder

File: frontend/src/microservices/content/objects/DeleteReviewPlannerRequest.ts
// DeleteReviewPlannerRequest：content 域删除评论请求对象。

export type DeleteReviewPlannerRequest = {
  userId: string
  reviewId: string
}

export const deleteReviewPlannerRequestFromJson = (json: string): DeleteReviewPlannerRequest =>
  JSON.parse(json) as DeleteReviewPlannerRequest

export const deleteReviewPlannerRequestToJson = (value: DeleteReviewPlannerRequest): string =>
  JSON.stringify(value)
